framework/infrastructure/traffic_director.py

```diff
@@ -679,32 +719,52 @@ def find_unused_forwarding_rule_port(
         # TODO(sergiitk): custom exception
         raise RuntimeError("Couldn't find unused forwarding rule port")
 
-    def create_forwarding_rule(self, src_port: int):
-        name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+    def create_forwarding_rule(self, src_port: int, ipv6=False):
+        if ipv6:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME + "-v6")
+            target_proxy = self.target_proxy_v6
+        else:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            target_proxy = self.target_proxy
+        
         src_port = int(src_port)
         logging.info(
             'Creating forwarding rule "%s" in network "%s": 0.0.0.0:%s -> %s',
             name,
             self.network,
             src_port,
-            self.target_proxy.url,
+            target_proxy.url,
         )
         resource = self.compute.create_forwarding_rule(
-            name, src_port, self.target_proxy, self.network_url
+            name, src_port, target_proxy, self.network_url
         )
-        self.forwarding_rule = resource
+        
+        if ipv6:
+            self.forwarding_rule_v6 = resource
+        else:
+            self.forwarding_rule = resource
+        
         return resource
 
-    def delete_forwarding_rule(self, force=False):
+    def delete_forwarding_rule(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            else:
+                name = self.make_resource_name(self.FORWARDING_RULE_NAME + "-v6")
         elif self.forwarding_rule:
             name = self.forwarding_rule.name
+        elif self.forwarding_rule_v6:
+            name = self.forwarding_rule_v6.name
         else:
             return
```

Comment:
wrap in `if ipv6` like the other cleanups.

Reply:
Done

Reply:
I don't see the change.

Reply:
Pushed a new commit. It was overridden due to interchange between cloudtop and local